frontend/inst_pointer_chase_gen_test.py

```diff
@@ -0,0 +1,115 @@
+def _PopNextFunction(function_list):
+    """Pops and returns the next function in the list."""
+    if not function_list:
+        raise IndexError("in _PopNextFunction: list of functions is empty")
+    return function_list.pop(0)
+
+
+class InstPointerChaseGeneratorTest(unittest.TestCase):
+
+    def setUp(self):
+        self.depth = 3
+        self.num_callchains = 2
+        self.gen = inst_pointer_chase_gen.InstPointerChaseGenerator(
+            self.depth, self.num_callchains, function_selector=_PopNextFunction)
+
+    def test_generate_callchain_mappings(self):
+        self.gen._GenerateCallchainMappings()
+        self.assertEqual(self.gen._caller2callee[0], 1)
+        self.assertEqual(self.gen._caller2callee[1], 2)
+        self.assertEqual(self.gen._caller2callee[2],
+                         inst_pointer_chase_gen.NO_CALLEE)
+        self.assertEqual(self.gen._caller2callee[3], 4)
+        self.assertEqual(self.gen._caller2callee[4], 5)
+        self.assertEqual(self.gen._caller2callee[5],
+                         inst_pointer_chase_gen.NO_CALLEE)
+
+    def test_generate_callchain_function_codeblockbodies(self):
+        self.gen._GenerateCallchainMappings()
+        self.gen._GenerateCallchainFunctions()
+        self.assertEqual(len(self.gen._functions),
+                         self.depth * self.num_callchains)
+
+        # All functions should use the same main code block body.
+        code_block_bodies = set()
+        for func_id, func in self.gen._functions.items():
+            code_block_bodies.add(func.instructions[0].code_block_body_id)
+            sig_body = (
+                self.gen._code_block_bodies[func.signature.code_block_body_id])
+            self.assertIn(self.gen._FunctionName(func_id),
+                          sig_body.instructions)
+            self.assertEqual(func.instructions[0].terminator_branch.type,
+                             cfg_pb2.Branch.BranchType.FALLTHROUGH)
+        self.assertEqual(
+            len(code_block_bodies),
+            1,
+            msg="All functions should share the same main code block body.")
+
+    def test_generate_callchain_function_callees(self):
+        self.gen._GenerateCallchainMappings()
+        self.gen._GenerateCallchainFunctions()
+        # Map from a caller function to its callee in the chain.
+        expected_caller2callee = {0: 1, 1: 2, 3: 4, 4: 5}
+
+        for func_id, func in self.gen._functions.items():
+            if func_id in expected_caller2callee:
+                # This function calls another in a chain.
+                self.assertEqual(len(func.instructions), 2)
+                self.assertEqual(func.instructions[1].terminator_branch.type,
+                                 cfg_pb2.Branch.BranchType.DIRECT_CALL)
+                expected_callee_func = (
+                    self.gen._functions[expected_caller2callee[func_id]])
+                self.assertEqual(
+                    func.instructions[1].terminator_branch.targets[0],
```

Comment:
These 1s and 0s in the indexes as well as the stream of dots are making this hard to read. Easiest solution I see is replace with a helper function local to this file.
e.g. branch_target_for_func(function)

Reply:
Done.